code/datums/components/durability.dm

```diff
@@ -0,0 +1,189 @@
+/// A component attached to any item (typically a weapon) that causes it to break after a certain number of uses.
+///
+/// Upon breaking, a number of effects can be applied to the item.
+/datum/component/durability
+	/// Whether the item has broken.
+	var/broken = FALSE
+
+	/// The prefix applied to the item when it breaks.
+	var/broken_prefix = "broken"
+
+	/// An absolute name to apply to the item when it breaks.
+	var/broken_name
+
+	/// An absolute description to apply to the item when it breaks.
+	var/broken_desc
+
+	/// The maximum number of times the item can be used before breaking.
+	var/max_durability
+
+	/// The current durability of the item. Upon reaching zero, the item breaks.
+	var/current_durability
+
+	/// If present, the icon state to apply to the item after it breaks.
+	var/broken_icon_state
+
+	/// If present, a sound that plays when the item breaks.
+	var/break_sound
+
+	/// The force of the item after breaking.
+	var/broken_force
+
+	/// A flat penalty to apply to the force of the item after breaking.
+	var/broken_force_decrease
+
+	/// The throw force of the item after breaking.
+	var/broken_throw_force = 0
+
+	/// A flat penalty to apply to the throw force of the item after breaking.
+	var/broken_throw_force_decrease = 0
+
+	/// The throw range of the item after breaking.
+	var/broken_throw_range
+
+	/// The embedding attributes of the item after breaking.
+	var/list/broken_embedding
+
+	/// The block chance of the item after breaking.
+	var/broken_block_chance
+
+	/// The `w_class` of the item after breaking.
+	var/broken_w_class
+
+	/// If present, a message that plays when the item breaks.
+	var/broken_message
+
+/datum/component/durability/Initialize(
+	broken,
+	broken_prefix,
+	broken_name,
+	broken_desc,
+	broken_embedding,
+	max_durability,
+	worn_durability,
+	current_durability,
+	worn_icon_state,
+	broken_icon_state,
+	break_sound,
```

Comment:
Frankly a lot more of these could also have default properties.
More default properties is good because it makes it easier for admins to stick the component on random objects.